audinterface/core/feature.py

```diff
@@ -468,58 +557,27 @@ def _values_to_frame(
             win_dur = None
             hop_dur = None
 
-        if self.process.process_func_is_mono and self.num_channels > 1:
-            features = np.concatenate(features)
-
-        if not isinstance(features, np.ndarray):
-            raise RuntimeError(
-                "Features must be a 'np.ndarray', "
-                f"not '{type(features)}'."
-            )
-
-        # features = np.array(features)
-        if features.ndim < 2 or features.ndim > 3:
-            raise RuntimeError(
-                f'Dimension of extracted features must be 2 or 3, '
-                f'not {features.ndim}.'
-            )
-
-        # Force third time step dimension
-        features = np.atleast_3d(features)
-        n_channels = features.shape[0]
-        n_features = features.shape[1]
-        n_time_steps = features.shape[2]
-
-        if n_channels != self.num_channels:
-            raise RuntimeError(
-                f'Number of channels must be {self.num_channels}, '
-                f'not {n_channels}.'
-            )
-        if n_features != len(self.feature_names):
-            raise RuntimeError(
-                f'Number of features must be {len(self.feature_names)}, '
-                f'not {n_features}.'
-            )
+        features = self._assert_shape_3d(features)
+        n_channels, n_features, n_frames = features.shape
 
         # Reshape features and store channel number as first feature
         # [n_channels, n_features, n_time_steps] =>
         # [n_channels * n_features + 1, n_time_steps]
```

Comment:
Looking at the code the result should be
```
# [n_channels * n_features, n_time_steps]
```
and maybe also replace `n_time_steps` by `n_frames` as in the code?

Reply:
As we can also clearly read from the code line how the new shape looks like it might also be sufficient if the comment just reads:
```python
# Reshape features and store channel number as first feature
```

Reply:
Actually the comment is also misleading. I think in a very first draft of `Features` we had the idea to encode the channel number as a feature, but then we changed it and instead we encode it in the column name of the `pd.DataFrame`. I now changed the comment to:

```
# Combine features and channels into one dimension
# f1-c1, f2-c1, ..., fN-c1, ..., f1-cM, f2-cM, ..., fN-cM
```

Reply:
Yes, much better, thanks.